src/games/strategy/triplea/ai/proAI/util/ProTransportUtils.java

```diff
@@ -162,6 +162,25 @@ public int findUnitsTransportCost(final List<Unit> units) {
     return transportCost;
   }
 
+  public List<Unit> getAirThatCantLandOnCarrier(final PlayerID player, final Territory t, final List<Unit> units) {
+    final GameData data = ai.getGameData();
+    int capacity = AirMovementValidator.carrierCapacity(units, t);
+    final Collection<Unit> airUnits = Match.getMatches(units, ProMatches.unitIsAlliedAir(player, data));
+    List<Unit> airThatCantLand = new ArrayList<Unit>();
+    for (final Unit airUnit : airUnits) {
+      final UnitAttachment ua = UnitAttachment.get(airUnit.getType());
+      final int cost = ua.getCarrierCost();
+      if (cost != -1) {
```

Comment:
Its not in just 2 or 3 places. Its everywhere (100s). Search the code base for -1 and you'll see that there are lots of parameters that use this as a default to indicate something isn't possible.

So  your code snippet is much cleaner and OO but would require a lot of refactoring to ever make happen and all the existing carrier capacity logic uses this method:
int capacity = AirMovementValidator.carrierCapacity(units, t);

To make any of these changes it would require massive refactoring or this one method would be very inconsistent with the rest of the code base. That's why I'm not gonna change it right now.